### PR TITLE
Deploy task update

### LINKED ISSUE #333: Combine installApps and deploy goals
Combine installApps and deploy goals and remove installApps. See equivalent ci.maven issue for details https://github.com/OpenLiberty/ci.maven/issues/506

### PR DESCRIPTION
**Pull request by mattbsox**
- Combined `deploy` and `installApps` logic
- Refactored `undeploy` task
- Added `deploy` and `undeploy` extensions

Still need to update deploy/undeploy docs with the new changes

Fixes #333

### CODE REVIEW

File: src/test/resources/sample.servlet/testLooseApplicationWithWarTask.gradle [hunk 1]
```diff
@@ -48,40 +48,32 @@ repositories {
...
+        serverXmlFile = file("src/main/liberty/config/server-apps-test.xml")
         packageLiberty {
             packageDirectory = packageDir
             packageName = appName
             include = packagingType
         }
-    	apps = [war1, war2]
-      verifyAppStartTimeout = 30
+        deploy {
+            apps = [war]
```
Comment (cherylking) [inline src/test/resources/sample.servlet/testLooseApplicationWithWarTask.gradle]: Is this still testing two apps? I don't really understand the changes here. And I see a different serverXmlFile referenced above.

**Comment by cherylking** [comment]
@ericglau Can you look into this?

File: src/main/groovy/io/openliberty/tools/gradle/Liberty.groovy [hunk 2]
```diff
@@ -176,6 +179,12 @@ class Liberty implements Plugin<Project> {
         }
     }
 
+    private static setDeployExtension(Project project) {
```
[the remark this answers is not in the document]
Comment (mattbsox) [reply src/main/groovy/io/openliberty/tools/gradle/Liberty.groovy]: Don't think so. We aren't using it to configure any plugin logic like we are with the DeployExtension.

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy [hunk 3]
```diff
@@ -15,68 +15,477 @@
...
+    protected void verifyAppStarted(String appFile, String appsDir) throws GradleException {
+        if (shouldValidateAppStart()) {
+            String appName = appFile.substring(0, appFile.lastIndexOf('.'))
+            if (appsDir.equals("apps")) {
+                ServerConfigDocument scd = null
+
+                File serverXML = new File(getServerDir(project).getCanonicalPath(), "server.xml")
+
+                try {
+                    scd = ServerConfigDocument.getInstance(CommonLogger.getInstance(), server.serverXML, server.configDirectory,
```
[the remark this answers is not in the document]
Comment (mattbsox) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy]: Yeah, missed that. Should be using serverXML. I'll fix it.

File: src/test/resources/liberty-test/oldStyleTest.gradle [hunk 4]
```diff
@@ -27,10 +27,6 @@ dependencies {
 liberty {
     serverName ='LibertyProjectServer'
 
-    deploy {
```
[the remark this answers is not in the document]
Comment (mattbsox) [reply src/test/resources/liberty-test/oldStyleTest.gradle]: This test doesn't have a server configured so we don't have anything to deploy.

File: src/test/resources/sample.servlet/testLooseApplicationWithWarTask.gradle (hunk 1 above)
Comment (mattbsox) [reply src/test/resources/sample.servlet/testLooseApplicationWithWarTask.gradle]: No it's not. This test wasn't configured correctly. The server.xml was wrong and the tests were checking for one app.

**Pull request merged**